pkg/translate/translate.go

```diff
@@ -526,3 +576,77 @@ func mergeK8sObject(base *object.K8sObject, overlayNode interface{}, path util.P
 
 	return newObj, nil
 }
+
+// createPatchObjectFromPath constructs patch object for node with path, returns nil object and error if the path is invalid.
+// eg. node:
+//     - name: NEW_VAR
+//       value: new_value
+// and path:
+//       spec.template.spec.containers.[name:discovery].env
+//     will constructs the following patch object:
+//       spec:
+//         template:
+//           spec:
+//             containers:
+//             - name: discovery
+//               env:
+//               - name: NEW_VAR
+//                 value: new_value
+func createPatchObjectFromPath(node interface{}, path util.Path) (map[string]interface{}, error) {
```

Comment:
would it be  better to move this to tpath or some other util file?

Reply:
I considered based on the function name but on closer inspection this is only used in this file and doesn't contain any functionality that would be useful to export at this time.